in-market-test/helper-nodes/setup_instructions.md

```diff
@@ -1,43 +1,86 @@
 
 ## Step 1: Download code and run a sample test
 1. If you do not have Rust installed on your machine, you can follow [these](https://doc.rust-lang.org/book/ch01-01-installation.html#installation) instructions to do so.
 2. Download the code from https://github.com/private-attribution/ipa
-   You can do this by running the command on the terminal: 
+   You can do this by running the command on the terminal:
    `git clone https://github.com/private-attribution/ipa`
 3. This will create a folder named “ipa”. Go inside the folder.
-   
+
    Command: `cd ipa`
-4. Build the Rust code 
-   
+4. Build the Rust code
+
    Command: `cargo build`
-5. Running a sample test : 
-   
+5. Running a sample test :
+
    Command: `cargo bench --bench oneshot_ipa --features="enable-benches" --no-default-features`
-   
+
 ## Step 2: Setup helper configuration
 1. Provide details of how the machine can be accessed from public internet. For this, we would need to know
    i.  IP address/DNS which is accessible from public internet (i.e. firewall allows connecting to the helper port (443))
-   ii. Confirm if machine is able to listen on port 443 
+   ii. Confirm if machine is able to listen on port 443
+
+2. Download docker image of IPA from shared folder. It we will be uploading an executable docker image which will have IPA code.
+   Link: TBD.  Alternatively, you can build the docker image directly from the Rust code with this command
+
+   `scripts/helper-image.sh --identity 1 --hostname localhost 1`
+
+3. Docker image already contains (default) public and private keys for encrypting match keys and also,
+   to secure the report files while in transit.
+
+   1. Installing/Upgrading Docker
+       - If you need to install docker, follow the instructions [here](https://docs.docker.com/engine/install/).
+       - If you already have docker, ensure that you are running at least Docker Version 20.10.22.
+          You can check version by running command `docker -v` on the terminal.
+
+   2. Create a docker image of the helper
+
+       `cd <path_to_ipa_source_code>`
+
+       `scripts/helper-image.sh --identity 1 --hostname localhost`
+
+   3.  Check which docker images are available to run and what is its TAG, run following command:
+
+       `docker images`
+
+       You should be able to see "private-attribution/ipa" in the list along with its TAG
+
+   3. To run the docker container with a bash terminal and see what is inside the image run:
+
+       `docker run -it docker.io/private-attribution/ipa:<IMAGE_TAG> /bin/bash`
+
+   4. To copy a TLS public key from the docker image to the host :     Open a separate terminal and run following command to see available docker images
+
+       `docker container ls`
+
+       Copy the CONTAINER ID corresponding to your IMAGE.
+
+       You will know what is your Helper party number i.e. 1,2 or 3. Here we show for Helper `1`:
+
+      `docker cp <CONTAINER_ID>:/etc/ipa/pub/1.pem .  `
```

Comment:
Just realized, we now have encryption keys also available. Could you add a line here for copying public key for encryption also?